File: my_cmake_package/src/my_talker.cpp
#include "my_talker.hpp"

MyTalkerNode::MyTalkerNode() : node('my_talker_node'), count_(0)
{
    publisher_ = this->create_publisher<std_msgs::msg::String>("/my_string_cpp", 10);

    timer_ = this->create_wall_timer(
        std::chrono::milliseconds(500),
        std::bind(&MyTalkerNode::talker_callback, this)
    );

    
}

void MyTalkerNode::talker_callback(){
    count_ ++;
    auto message = std_msgs::msg::String();
    message.data = "Hello, world! " + std::to_string(count_++);
    RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
    publisher_->publish(message);
}

void MyTalkerNode::main_loop()
{


}

int main(int argc, char *argv[]) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyTalkerNode>();

    // Instead of spin, use a custom executor
    rclcpp::executors::SingleThreadedExecutor exec;
    exec.add_node(node);
    exec.spin();

    rclcpp::shutdown();
    return 0;
}

File: my_python_package/my_python_package/my_talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MySubriberNode (Node):
    def __init__(self):
        super().__init__("my_talker_node")
        self.counter = 0
        self.publisher_ = self.create_publisher(String, '/my_string',  10)
        self.timer_ = self.create_timer(1.0, self.timer_callback)
        # self.get_logger().info('My subriber node has started and is listening to /my_string')

    def timer_callback (self):
        self.counter +=1
        msg = String()
        msg.data = f'Publishing message number {self.counter}'
        # self.get_logger().info(f'Received: {msg.data}')
        self.publisher_.publish((msg))

def main (args=None):
    rclpy.init(args=args)
    node = MySubriberNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ =='__main__':
    main()